docs/Contributing-back.rst

```diff
@@ -0,0 +1,496 @@
+
+.. code:: shell
+
+  git checkout -b mybranch
+
+You can choose the branch name freely, but make it sufficiently descriptive --
+your future self will thank you.
+
+
+.. _contributing_back_creating_pull_requests_commit:
+
+Committing your changes
++++++++++++++++++++++++
+
+To 'save' your changes, you should create one or more *commits* in the branch
+you created. It is recommended to make separate commits for different 'units of work'.
+
+First, make sure you have checked out the branch were you want to commit the changes to.
+
+For example, to commit changes to the ``mybranch`` branch:
+
+.. code:: shell
+
+  git checkout mybranch
+
+To check which branch is currently checked out, use ``git branch``.
+
+To get a high-level overview of the changes before committing them, you can use ``git status``.
+
+To see the actual changes that were made, use ``git diff``.
+
+To commit the changes you want to contribute back, use ``git add <files>``
+to *stage* the changes, followed by ``git commit -m "<message>"`` to create
+the actual commit.
+
+For example, to commit the changes that were made to ``easybuild/tools/filetools.py``:
+
+.. code:: shell
+
+  git add easybuild/tools/filetools.py
+  git status  # check which files are staged for commit
+  git commit -m "example commit message for changes to filetools.py"
+
+.. note:: Please use a concise commit message that describes the changes you made.
+
+To create a single commit for *all* the changes that were made, use ``git commit -am "<message"``.
+
+For example:
+
+.. code:: shell
+
+  # first, check changes to make sure we want to commit them all at once
+  git status
+  git diff
+
+  git commit -am "example commit message for all the changes that were made"
+
+To verify that your work was committed, use ``git log`` to see all commits
+on the current branch. Use ``git log --stat`` and/or ``git log --diff`` to see
+more details about which changes are included in each of the commits.
+
+
+.. _contributing_back_creating_pull_requests_push:
+
+Pushing your branch
++++++++++++++++++++
+
+Once you have committed your changes to a branch, you should *push* your
+branch to your fork of the GitHub repository using ``git push``.
+
+For example, to push the ``mybranch`` branch to the ``myfork`` remote GitHub repository:
+
+.. code:: shell
+
+  git push myfork mybranch
+
+Note that this is equivalent to making your work public.
+
+
+.. _contributing_back_creating_pull_requests_open_pr:
+
+Opening the pull request
+++++++++++++++++++++++++
+
+To open a pull request using the branch you pushed,
+you should use the GitHub interface, and follow the steps outlined below.
+
+.. note:: Replace '``EXAMPLE``' with your GitHub account name, and
+          '``easybuild-framework``' with the name of the target EasyBuild repository.
+
+i. visit https://github.com/EXAMPLE/easybuild-framework;
+ii. switch to the branch that includes the changes you want to contribute back
```

Comment:
Mention explicitly `mybranch`. All the buttons/options are highlighted, except this one.